### ISSUE TITLE
[BUG] Reproduce HDN paper performance

### ISSUE THREAD

**Issue by veegalinova** [CAREamics/careamics#602]
## Describe the bug

Currently, BSD archives inferior performance from what was reported in the paper. With current defaults, the validation performance is ~**23** PSNR vs **28.70** reported in the paper. 

## To Reproduce
The current defaults are trained with: 
```python
config = create_hdn_configuration(
    experiment_name=experiment_name,
    data_type="tiff",
    axes="SYX",
    patch_size=(64, 64),
    batch_size=64,
    num_epochs=50
)
train_data_module = TrainDataModule(
    data_config=config.data_config,
    train_data=train_path,
    val_data=val_path
)
trainer = Trainer(max_epochs=50)
trainer.fit(model, datamodule=train_data_module)
```

## System

> - Python version 3.12.2
> - PyTorch version 2.8.0
> - PyTorch lightning version 2.5.5
> - CAREamics version - current main (as for end Oct 2025)

**Comment by veegalinova** [comment]
Current progress: 

There are some parameter discrepancies between the original implementation repo (BSD training taken from [here](https://github.com/juglab/HDN/blob/main/examples/Pixel_Noise/Natural/1-Training.ipynb)) and the current defaults. 
Currently, the net has `970,785` params, and the original net had `7,301,889` params (why so many?!) 
Here is a list of model parameters that equalizes the number of parameters and the layers between implementations: 

```python
model.encoder_n_filters = 64
model.decoder_n_filters = 64
model.z_dims = [32]*6
config.algorithm_config.model.nonlinearity = "ELU"
config.algorithm_config.predict_logvar = None  
model.encoder_dropout = 0.2
model.decoder_dropout = 0.2

# Hard coded parameters that needed changing
model.encoder_blocks_per_layer = 5
model.decoder_blocks_per_layer = 5
model.first_conv_kernel = 5 
```

The training parameters in the paper are like follows: 
```python
data_config.patch_size = (128, 128)
data_config.batch_size = 16

algorithm_config.num_epochs = 400  # with steps per epoch = 500! 
algorithm_config.optimizer.parameters = {'lr': 3e-4}
algorithm_config.loss.kl_params.free_bits_coeff = 1.0
algorithm_config.lr_scheduler.parameters = {'patience': 10, 'factor': 0.5, 'min_lr': 1e-12}
```
\+ the original was trained with `AdamW`, `mmse_count` for prediction was `100`.

**Comment by veegalinova** [comment]
Experiments on Convallaria dataset with a noise model: 

The parameters in the original implementation are almost the same: 
```python
config.data_config.patch_size = (64, 64)
config.data_config.batch_size = 64

# Training 
config.algorithm_config.num_epochs = 400  # with reduced steps per epoch! 
config.algorithm_config.optimizer.parameters = {'lr': 3e-4}
config.algorithm_config.loss.kl_params.free_bits_coeff = 1.0
config.algorithm_config.lr_scheduler.parameters = {'patience': 10, 'factor': 0.5, 'min_lr': 1e-12}


# Model 
config.algorithm_config.model.encoder_n_filters = 64
config.algorithm_config.model.decoder_n_filters = 64
config.algorithm_config.model.z_dims = [32]*6
config.algorithm_config.model.encoder_dropout = 0.2
config.algorithm_config.model.decoder_dropout = 0.2
config.algorithm_config.model.nonlinearity = "ELU"
config.algorithm_config.predict_logvar = "pixelwise"  
config.algorithm_config.noise_model = MultiChannelNMConfig(noise_models=[nm_config])

# hardcoded
model = VAEModule(config.algorithm_config)
model.model.encoder_blocks_per_layer = 5
model.model.decoder_blocks_per_layer = 5
```

There is problem thought, with patch size 64, the image becomes too small somewhere in the middle of the network and fails. Fixed by reducing the number of latents (`z_dims` set to default). After that, the validation metric (at mmse 100) is almost the same as in the paper: 
```
PSNR: 37.16 +/- 0.04
Reported PSNR: 37.39
```

**Comment by veegalinova** [comment]
final parameter set for BSD: 

all of the above, plus 
```python
config.algorithm_config.loss.reconstruction_weight = 6.421740 # estimation based on noise level in the dataset 
config.algorithm_config.predict_logvar = None # because we do not use a noise model
```
The code for reproducibility is temporarily in a fork: https://github.com/veegalinova/careamics/blob/hdn_debugging/dev/HDN_debug/experiments/train_bsd.py 

metrics (the reported PSNR is with a fixed range of 255): 
```bash
PSNR: 28.93 +/- 2.52
Reported PSNR: 28.82
```

Next steps: 
- plot how mmse count influences the metric, try to find an optimal value (that is lower than 100 in the baseline)
- refactor VAE lightning module (to work with NG dataset and in general)
- compare metrics with NG dataset
- more parameter ablations
- try other datasets reported in the paper
